tfx/components/example_gen/utils_test.py

```diff
@@ -664,6 +666,100 @@ def testCalculateSplitsFingerprintSpanAndVersionWithDate(self):
     self.assertEqual(splits[0].pattern, '19700103/ver02/split1/*')
     self.assertEqual(splits[1].pattern, '19700103/ver02/split2/*')
 
+  def testRangeConfigWithNonexistentSpan(self):
+    # Test behavior when specified span in RangeConfig does not exist.
+    span1_split1 = os.path.join(self._input_base_path, 'span01', 'split1',
+                                'data')
+    io_utils.write_string_file(span1_split1, 'testing11')
+
+    range_config = range_config_pb2.RangeConfig(
+        static_range=range_config_pb2.StaticRange(start_span_number=2,
+                                                  end_span_number=2))
+    splits = [
+        example_gen_pb2.Input.Split(
+            name='s1', pattern='span{SPAN:2}/split1/*')
+    ]
+
+    with self.assertRaises(tf.errors.NotFoundError):
```

Comment:
why it's notFoundError instead of ValueError

Reply:
b/c the glob pattern generated wont exist on file system, so iirc tf.gfile.glob will return a notFoundError.